wurst/systems/spells/TrackerBeacon.wurst

```diff
@@ -0,0 +1,58 @@
+package TrackerBeacon
+import HashMap
+import GameTimer
+import RegisterEvents
+import UnitExtensions
+import SimError
+import PlayerExtensions
+import PingMinimap
+
+
+let SPELL_BEACON_CAST = 'A05M'
+let SPELL_BEACON_QUERY = 'A05N'
+let BEACON_DURATION = 5 * 60
+
+
+class Beacon
+    player target
+    real timestamp
+
+    construct(player target)
+        this.target = target
+        this.timestamp = getElapsedGameTime()
+
+let beacons = new HashMap<player, Beacon>()
+
+function castBeacon()
+    let target = GetSpellTargetUnit()
+    let owner = GetSpellAbilityUnit().getOwner()
+
+    if not target.isTroll()
+        simError(owner, "Can't beacon that unit")
+        return
+
+    if beacons.has(owner)
+        beacons.remove(owner)
+
+    let beacon = new Beacon(target.getOwner())
+    beacons.put(owner, beacon)
+
+function queryBeacon()
+    let owner = GetSpellAbilityUnit().getOwner()
+
+    if not beacons.has(owner)

```

Comment:
Can you use a lookup without checking this and then compare the result to `null`?